### PR TITLE
EQL: Remove match functions

### LINKED ISSUE #63263: EQL: Remove `match` function
Currently `match` function is used to perform regex-like string matching. As this functionality is under development, remove the current function from 7.10.

### PR DESCRIPTION
**Pull request by costin**
Since match (for matching regex) is not currently in use remove it for
now.

Close #63263

### CODE REVIEW

**Review by astefan — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt [hunk 1]
```diff
@@ -304,39 +304,38 @@ InternalEqlScriptUtils.cidrMatch(InternalQlScriptUtils.docValue(doc,params.v0),p
...
+//"regexp":{"command_line":{"value":"^.*?net.exe|net\\.exe|C:\\\\Windows\\\\system32\\\\net1\\s+"
+//;
+//
+//matchFunctionScalar
+//process where match(substring(command_line, 5), "^.*?net.exe", "net\\.exe", "C:\\\\Windows\\\\system32\\\\net1\\s+")
+//;
+//"script":{"source":"InternalQlScriptUtils.nullSafeFilter(InternalSqlScriptUtils.regex(InternalEqlScriptUtils.substring(
+//InternalQlScriptUtils.docValue(doc,params.v0),params.v1,params.v2),params.v3))",
+//"params":{"v0":"command_line","v1":5,"v2":null,"v3":"^.*?net.exe|net\\.exe|C:\\\\Windows\\\\system32\\\\net1\\s+"}}
+//;
```
Comment (bpintea) [inline x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt:331]: did you mean to still keep these around?

**Review by bpintea — APPROVED** [review]
LGTM

File: x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt (hunk 1 above)
Comment (costin) [reply x-pack/plugin/eql/src/test/resources/queryfolder_tests.txt:331]: Yes since match is only temporarily removed.

**Pull request merged**